lib/hawkular/operations/operations_api.rb

```diff
@@ -215,7 +215,10 @@ def check_pre_conditions(hash = {}, params = [], &callback)
       fail 'callback must have the perform method defined. include Hawkular::Operations' unless
           callback.nil? || callback.respond_to?('perform')
       params.each do |property|
-        fail "Hash property #{property} must be specified" if hash[property].nil?
+        if hash[property].nil?
+          error_message = "Hash property #{property} must be specified"
+          callback.nil? ? fail(ArgumentError, error_message) : callback.perform(:failure, error_message)
```

Comment:
While you are right @abonas the situation is even more different as the `callback.perform` works on the failure callback, but that is not checked for. 

The only situation this method should `fail` is if no error-callback is given. Otherwise the error-callback should be called with the problem (message).